lib/model-builder.js

```diff
@@ -342,6 +342,32 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     });
   }
 
+  // updateOnly property is added to indicate that this property will appear in
+  // the model for update/updateorcreate operations but and not for create operation.
+  var forceId = ModelClass.settings.forceId;
+  if (idNames.length > 0) {
+    var idName = modelDefinition.idName();
+    idProp = ModelClass.definition.rawProperties[idName];
+    if (idProp.generated && forceId !== false) {
+      forceId = 'auto';
+    } else if (!idProp.generated && forceId === 'auto') {
+      // One of our parent models has enabled forceId because
+      // it uses an auto-generated id property. However,
+      // this particular model does not use auto-generated id,
+      // therefore we need to disable `forceId`.
+      forceId = false;
+    }
+
+    if (forceId) {
+      ModelClass.validatesAbsenceOf(idName, {if: 'isNewRecord'});
```

Comment:
@rashmihunt @bajtos shouldn't we also reject any requests containing an empty string ID when `forceId` is `true`? [`validatesAbsenceOf`](/strongloop/loopback-datasource-juggler/blob/master/lib/validations.js#L298) considers an [empty string value](/strongloop/loopback-datasource-juggler/blob/master/lib/validations.js#L742) valid, which generates a record with no id (at least with MongoDB) when calling a POST endpoint with an explicit empty string ID even if `forceId` is `true` (I'm new to LoopBack, so I might have missed something).

Reply:
For posterity, I responded here: https://github.com/strongloop/loopback-datasource-juggler/issues/1519#issuecomment-364513700